fstest/fstests/fstests.go

```diff
@@ -1407,6 +1407,63 @@ func Run(t *testing.T, opt *Opt) {
 
 		})
 
+		// TestPutUnknownSize ensures Fs.Put() and Object.Update() don't panic when
+		// src.Size() == -1
+		t.Run("FsPutUnknownSize", func(t *testing.T) {
+
+			testPutUnknownSize := func() {
```

Comment:
These two functions could be separate tests, or they could be subtests

    t.Run("Put", func(t*testing.T) {
    t.Run("Update", func(t*testing.T) {